test/GitHub.App.UnitTests/Services/RepositoryCloneServiceTests.cs

```diff
@@ -122,6 +122,23 @@ public async Task UpdatesMetricsWhenDefaultClonePath(string targetPath, string d
                     ((MemberExpression)x.Body).Member.Name == counterName));
         }
 
+        [Test]
+        public async Task CleansDirectoryOnCloneFailed()
+        {
+            var serviceProvider = Substitutes.GetServiceProvider();
+            var operatingSystem = serviceProvider.GetOperatingSystem();
+            var vsGitServices = serviceProvider.GetVSGitServices();
+            var cloneService = CreateRepositoryCloneService(serviceProvider);
+
+            Assert.ThrowsAsync<Exception>(async () => {
+                await cloneService.CloneRepository("https://github.com/failing/url", @"c:\dev\bar");
+            });
+
+            operatingSystem.Directory.Received().CreateDirectory(@"c:\dev\bar");
+            operatingSystem.Directory.Received().DeleteDirectory(@"c:\dev\bar");
+            await vsGitServices.Received().Clone("https://github.com/failing/url", @"c:\dev\bar", true);
```

Comment:
Could you factor out "https://github.com/failing/url" and @"c:\dev\bar" into something like `cloneUrl` and `clonePath`?